### PR TITLE
Compound Search: Identify matched filter blocks

### PR DESCRIPTION
**Pull request by alexkb0009**
Result on local:

img

### CODE REVIEW

File: src/encoded/search/compound_search.py [hunk 1]
```diff
@@ -230,6 +237,7 @@ def execute_filter_set(context, request, filter_set, from_=0, to=10,
                 subreq = cls.build_subreq_from_single_query(request, query, route=cls.BUILD_QUERY_URL,
                                                             from_=from_, to=to)
                 sub_query = request.invoke_subrequest(subreq).json[cls.QUERY]
+                sub_query["bool"]["_name"] = block_index
```
Comment (alexkb0009) [inline src/encoded/search/compound_search.py:240]: Gets converted to a string it seems, hence converting back to `int` above.

File: src/encoded/search/compound_search.py [hunk 2]
```diff
@@ -101,11 +101,18 @@ def format_result_for_endpoint_response(request, es_results, filter_set, result_
         if search_builder_instance.search_session_id:  # Is 'None' if e.g. limit=all
             request.response.set_cookie('searchSessionID', search_builder_instance.search_session_id)
 
+        result_list = []
+        for hit in es_results['hits'].get("hits", []):
+            result = hit['_source']['embedded']
+            filter_block_indices = [ int(query_index) for query_index in hit.get("matched_queries", []) ]
+            result["__matching_filter_block_indices"] = filter_block_indices
+            result_list.append(result)
```
Comment (willronchetti) [inline src/encoded/search/compound_search.py:109]: Note in the docstring that `__matching_filter_block_indices` is set here.

File: src/encoded/search/compound_search.py (hunk 1 above)
Comment (willronchetti) [inline src/encoded/search/compound_search.py:240]: I would add a comment on this line linking to the feature documentation (https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-request-named-queries-and-filters.html)

**Review by willronchetti — APPROVED** [review]
Needs version bump, suggested adding some small comments just for clarity in case we need to update this in the future, otherwise LGTM

**Pull request merged**